File: solutions/python/hamming/1/hamming.py
def distance(strand_a, strand_b):
    """
    Calculating the Hamming Distance between two DNA strands.
    """
    if len(strand_a) != len(strand_b):
        raise ValueError("Strands must be of equal length.")

    return sum(a != b for a, b in zip(strand_a, strand_b))

File: solutions/go/hamming/1/hamming.go
package hamming

import "errors"

func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
        return 0, errors.New("length is different")
    }

	var distance = 0
	for i := 0; i < len(a); i += 1 {
        if a[i] != b[i] {
            distance += 1
        }
    }
	return distance, nil
}
